### DISCUSSION TITLE
How to use Kedro in a project without the "src" folder?

### DISCUSSION THREAD

**Discussion by jmeidam** [kedro-org/kedro#1836] [Idea]
I have an existing project where the python package-folder is located in the root of the repo, together with setup.py and requirements.txt and such.
After searching I can find older posts (2020) that mention I can add a .kedro.yml file in my repo with `source_dir: .`.
However, this does not seem to work. I still get an error when running kedro: 
`NotADirectoryError: Source path '<path-to-cloned-repo>/src' cannot be found`

Is there a different way to not have the src folder, but still use Kedro?

**Comment by noklam** [comment]
@jmeidam Can you share the link?

**Reply by noklam** [reply]
`kedro.yml` is replaced with `pyproject.toml` which is now  a Python standard.
```
#pyproject.toml
[tool.kedro]
package_name = "no_src"
project_name = "no_src"
project_version = "0.18.2"
source_dir = ""  # Add this line
```

**Reply by jmeidam** [reply]
Hi, the link to what? The repo in question is an internal project, but the folder structure is a bit like this:
```repo:
├── /.github/workflows       # Github workflows to automate processes such as running 
│                            # unit tests and building documentation
│
├── /adb_notebooks               # Databricks notebooks are stored in this folder as *.py files
|
├── /config             
|
├── /doc                         # Folder for Sphinx documentation
│
├── /tests                       # Folder to store the unit-tests
│
├── /project_package_name  # Package, not nested under src directory
│   ├── module.py                  # Example module
|   └── __init__.py              
|
├── .gitignore                   
├── MANIFEST.in                 
├── pytest.ini                   
├── README.md                   
├── requirements.txt        
├── requirements_dev.txt        
└── setup.py                     
```

**Reply by noklam** [reply]
@jmeidam I was referring to the posts that you found.

> After searching I can find older posts (2020) that mention I can add a .kedro.yml file in my repo with source_dir: ..

**Reply by jmeidam** [reply]
> `kedro.yml` is replaced with `pyproject.toml` which is now a Python standard.
> 
> ```
> #pyproject.toml
> [tool.kedro]
> package_name = "no_src"
> project_name = "no_src"
> project_version = "0.18.2"
> source_dir = ""  # Add this line
> ```

Ah thanks, that does indeed do the trick. So when I find a post that mentions the .kedro.yml file, I should incorporate that into the [tool.kedro] section of pyproject.toml

**Reply by jmeidam** [reply]
> @jmeidam I was referring to the posts that you found.
> 
> > After searching I can find older posts (2020) that mention I can add a .kedro.yml file in my repo with source_dir: ..

This is one of them:
https://github.com/kedro-org/kedro/issues/149#issuecomment-617186704
